packages/definitions-parser/src/lib/definition-parser.ts

```diff
@@ -53,17 +41,25 @@ function formattedLibraryVersion(typingsDataRaw: TypingsDataRaw): `${number}.${n
   return `${typingsDataRaw.header.libraryMajorVersion}.${typingsDataRaw.header.libraryMinorVersion}`;
 }
 
-export async function getTypingInfo(packageName: string, dt: FS): Promise<TypingsVersionsRaw | { errors: string[] }> {
+export async function getTypingInfo(
+  packageNameOrTypesDirectoryName: string,
```

Comment:
This name is a little unwieldy now and it doesn't look to me that the semantics have changed. If some callers always pass a directory name and some callers always pass a package name, maybe it would be worthwhile to have two entry points?

Reply:
The semantics didn’t need to change, because the packageName→directoryName transform is a noop on directory names, so we can just perform it unconditionally. I changed it because I noticed that I was passing a directory name in and wondered if that was going to break something.

Reply:
Yeah, I think my later suggestion to have two string types is the right approach actually.